### DISCUSSION TITLE
How to remove stylua from mason?

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [LazyVim/LazyVim#1185] [Q&A]
By default, Lazyvim will use mason to install `stylua`. However, the release package of `stylua` cannot be installed on Apline due the missing of `glibc`. 

Therefore, I would like to install `stylua` manually by `apk add`, and remove it from mason. How to remove it from mason's `ensured_installed`?

**Comment by dpetka2001** [comment]
The only way I could think of is to configure `Mason`'s `opts` as a function which returns a new table with the values that you want. This would overwrite Lazyvim's default `Mason` configuration. I'm not sure however if you use `vim.list_extend()` in some other place if it will extend your overwritten configuration or just try to extend the Lazyvim's parent configuration, thus making this obsolete. Maybe @folke could have a better opinion as how to achieve this.

**Comment by ChenZhongPu** [comment]
A feasible (not elegant) way:

```lua
    opts = function(_, opts)
      for i = #opts.ensure_installed, 1, -1 do
        if opts.ensure_installed[i] == "stylua" then
          table.remove(opts.ensure_installed, i)
        end
      end
    end,
```

**Reply by dpetka2001** [reply]
Thanks for posting this. Although I don't want to do it myself, it's useful to know how it can be done without overwriting default configuration.
